### PR TITLE
restore full set of configuration options

### PR DESCRIPTION
**Pull request by bloerwald**
image

### CODE REVIEW

**Comment by axc450** [comment]
Thanks for adding these back in. I loaded up the zip you sent me so some of the below comments might not be applicable (or the action is on me to fix) but thought Id better mention them anyway.

Heres what I currently see:
image

Comments:
- Can we have "Autobattle Hotkey" and the button inline rather then on seperate lines? A bit like how the sound option is
- Can we shorten "Play sound when..." it is quite long and therefore clipping
- I dont have any sounds
image
- Theres a weird hover tooltip effect going on (this should be removed)
image
- I assume "Debug: Action test break script" means "Stop executing script when a break statement is encountered". Can we get this renamed (or removed? is this useful to have?)
- Why does hiding the minimap icon require a UI reload?
image
- Is it worth evaluating if we need the settings (or certain settings) in the "Script selector" section? Having things more configurable is great, but should be limited to things users would actually find useful.

**Review by axc450 — CHANGES_REQUESTED** [review]
Please see above comments :)

**Comment by bloerwald** [comment]
- Keep in mind that all these changes are just restoring old functionality, not adding anything.
- "can we make this say x instead?": all of these are localised and can be changed in the curseforge interface. I did not change any of them from the originals since I didn't have access back then.

-----

> Can we have "Autobattle Hotkey" and the button inline rather then on seperate lines? A bit like how the sound option is

Sadly no. We use Ace and the control for keybinds has no ability to do that. I tried, really. This entire thing is quite optimised for filling exactly one page.

> Can we shorten "Play sound when..." it is quite long and therefore clipping

See translations.

> I dont have any sounds

This is impressive. Then again, neither this addon nor Ace supplies any sounds. We can make it add some "our" sound, but usually people have DBM/BigWigs, which provides a lot already. Not having any addon providing a sound is quite a challenge, actually.

> Theres a weird hover tooltip effect going on (this should be removed)

This is the mouse-over of the control below, and since this UI is done by Ace, we can't change it.

> I assume "Debug: Action test break script" means "Stop executing script when a break statement is encountered". Can we get this renamed (or removed? is this useful to have?)

See translations. 

I don't like this being a "debug" option, so I'd like to remove that prefix. Yes, it means that if `test()` is executed, it outputs and stops rather than outputting and continuing. 

> Why does hiding the minimap icon require a UI reload?

I have no idea, didn't change anything. The answer has to be laziness.

> Is it worth evaluating if we need the settings (or certain settings) in the "Script selector" section? Having things more configurable is great, but should be limited to things users would actually find useful.

Most users likely never use anything but Rematch. The other options are effectively dead, I don't know a single person using this addon without Rematch. Thus, all the data is dead, since only one option exists. As long as we *have* those options, I'd have the ability to reorder it though.

--------

tl;dr: I'd keep options as they are since they had been like that before. If you want to change the texts, I'm 100% up for that.

**Comment by bloerwald** [comment]
> Why does hiding the minimap icon require a UI reload?

Completely stupid reason, now no longer does. Additionally fixes #40 which is closely related.

**Comment by bloerwald** [comment]
No longer needs localisation string `OPTION_SETTINGS_HIDE_MINIMAP_TOOLTIP`. It can be removed from curse.

**Comment by axc450** [comment]
Thanks for the updates :) I have taken another look at this:
- I usually run _only_ PBS and Rematch when testing the addon, so that is why I didnt have any sounds. I enabled DBM but still didnt have any. Weakauras worked though, so I think its fine (does DBM not provide any sounds??). A small request: Can we have the sound play when one is selected? This is how it works in most addons. Currently, I can select a sound but only actually hear it within a pet battle.
- If text changes can all be done in Curse, Ill sort that out!
- Do we want to support this new Dragonflight addon menu thingy?
image
- I still think we should probably remove the "Script selector" section as 99% of people will be using Rematch (and the addon should just default to "First enemy" if Rematch isnt found. But your decision on this one 👍 
- Can #43 be closed?

Ill try and get this into the next release. Thanks for restoring the options page! 😄

**Comment by bloerwald** [comment]
> * I usually run _only_ PBS and Rematch when testing the addon, so that is why I didnt have any sounds. I enabled DBM but still didnt have any. Weakauras worked though, so I think its fine (does DBM not provide any sounds??). A small request: Can we have the sound play when one is selected? This is how it works in most addons. Currently, I can select a sound but only actually hear it within a pet battle.

The control has a bell icon on the left side. Clicking it previews the sound. While the dropdown is open, the same icon on the right side previews the sound.

> * Do we want to support this new Dragonflight addon menu thingy?

This is independent from this Pull Request. We can do that, if you want, but please handle it as a separate issue.

> * Can [Fix global leak #43](https://github.com/axc450/pbs/pull/43) be closed?

Yes, it is included here, so you can close it if this one is merged.

**Review by axc450 — APPROVED** [review]
